File: tools/csharp/FileProbe.cs
using System;
using System.IO;

public static class FileProbe
{
    private static string JsonEscape(string value)
    {
        return value
            .Replace("\\", "\\\\")
            .Replace("\"", "\\\"")
            .Replace("\n", "\\n")
            .Replace("\r", "\\r")
            .Replace("\t", "\\t");
    }

    public static int Main(string[] args)
    {
        if (args.Length != 1)
        {
            Console.Error.WriteLine("usage: mono fileprobe-cs.exe <file>");
            return 2;
        }

        try
        {
            var info = new FileInfo(args[0]);
            Console.WriteLine(
                "{{\"language\":\"csharp\",\"name\":\"{0}\",\"size\":{1}}}",
                JsonEscape(info.Name),
                info.Length
            );
            return 0;
        }
        catch (Exception error)
        {
            Console.Error.WriteLine(error.Message);
            return 1;
        }
    }
}

File: tools/cpp/file_probe.cpp
#include <cstdint>
#include <filesystem>
#include <iostream>
#include <string>

namespace fs = std::filesystem;

std::string json_escape(const std::string& value) {
    std::string escaped;
    for (char ch : value) {
        switch (ch) {
            case '\\':
                escaped += "\\\\";
                break;
            case '"':
                escaped += "\\\"";
                break;
            case '\n':
                escaped += "\\n";
                break;
            case '\r':
                escaped += "\\r";
                break;
            case '\t':
                escaped += "\\t";
                break;
            default:
                escaped += ch;
        }
    }
    return escaped;
}

int main(int argc, char** argv) {
    if (argc != 2) {
        std::cerr << "usage: fileprobe-cpp <file>\n";
        return 2;
    }

    fs::path path(argv[1]);
    std::error_code error;
    auto size = fs::file_size(path, error);
    if (error) {
        std::cerr << error.message() << "\n";
        return 1;
    }

    std::cout << "{\"language\":\"cpp\",\"name\":\""
              << json_escape(path.filename().string()) << "\",\"size\":" << size << "}\n";
    return 0;
}
